### PR TITLE
Port to py3.6 fixes #1252

### LINKED ISSUE #1252: port nipapd to python 3.6
In order to preserve this great app, I'd like to pitch in with some refurbishing of nipapd - I might do nipapwww also eventually..

* 2to3:
All the classics: print, exceptions, instanceof, 0<=mask<=32 support, remove unicode() etc etc
* optimize string formatting:
Py3 brings more readable str.format, but as performance is key, I'm going to only use it for exceptions.
Concatenation is by far the most performant for strings, so this is preferred and otherwise the % operator is used
For logging, %s allows params to be embedded only if the logging is not filtered out, so this is implemented to avoid unneeded execution of the % operator.
I'll harmonise a few things like consistently using " around ' and vice versa, and never \' or using ''' around '..
The way strings are being formatted has a big impact when done often, and since sql is generated often it will pay to make it more efficient.
* update libs / replace libs:
ConfigParser -> configparser
psycopg2 -> psycopg2-binary (and drop use of Inet class in backend)
flask-xml-rpc -> flask-xml-rpc-re (former is dead since 2010 and py2 only)
updated flask-compress import syntax

### PR DESCRIPTION
**Pull request by steffenschumacher**
Move to newest or replacement libs
Fix syntax to 3.6+
Overhaul strings to optimize performance and readability
Update build stuff to 3.6 + move docker img to 18.04/bionic

Disclaimer: I have no clue on the linux packaging stuff, but the docker img builds just fine..

### CODE REVIEW

File: nipap/nipap/backend.py [hunk 1]
```diff
@@ -547,63 +547,31 @@
...
             raise authlib.AuthorizationFailed("read-only user '%s' is not authorized to run function '%s'" % (auth.username, f.__name__))
         return f(*args, **kwargs)
 
     return decorated
 
 
-
-
-
-class Inet(object):
```
Comment (steffenschumacher) [inline nipap/nipap/backend.py:579]: Removed as its no longer needed..

File: nipap/xmlbench.py [hunk 2]
```diff
@@ -84,47 +85,49 @@ def makeRequest(self):
...
             self.makeRequest()
 
 
 if __name__ == '__main__':
     import argparse
+
     parser = argparse.ArgumentParser()
-    parser.add_argument('-p', '--port', dest='port', type='int', default='1337', help="TCP port")
+    parser.add_argument(
+        '-p', '--port', dest='port', type='int', default='1337', help="TCP port"
```
[the remark this answers is not in the document]
Comment (steffenschumacher) [reply nipap/xmlbench.py:121]: Its actually shorter now, however I was not aware of the 79 ch rule..

File: nipap/xmlbench.py [hunk 3]
```diff
@@ -1,65 +1,68 @@
...
         self.params = params
         self.start_time = 0
         self.end_time = 0
-        self.value =  ""
+        self.value = ""
         self.error = ""
         self.finished = False
-        self.error_file = open('errors.csv','w+')
-
+        self.error_file = open('errors.csv', 'w+')
```
[the remark this answers is not in the document]
Comment (steffenschumacher) [reply nipap/xmlbench.py:19]: I've not been aware of the 79 ch rule, and haven't paid much attention to the test scripts..

**Comment by garberg** [comment]
Thanks a lot for this great contribution! Before merging I'd like to make sure the CI tests pass. I'll investigate what's happened to Travis CI and/or run the tests locally, so please bare with me. Have you tried running the test suite locally?

**Comment by steffenschumacher** [comment]
Hi,
Yeah, I’m working on those actually, and I did find 2-3 issues after I migrated the tests to 3.6 also - if it’s easier I can do one big PR for all of it, but I figured chunks would be easier..

**Comment by steffenschumacher** [comment]
Oh yeah - I guess hound uses black, so I'm wondering if I should put all source files through black WRT formatting?
I think it might mess up your doc-style, and I'm not entirely sure I like the linebreaks its doing, but that's for you to decide..
IMHO its not really worth the effort, but its not my project either..

**Comment by garberg** [comment]
Merged in #1316 after some further fixes.

**Pull request closed without merging**